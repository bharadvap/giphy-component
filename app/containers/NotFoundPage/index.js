/**
 * NotFoundPagePage
 *
 * This is the page we show when the user visits a url that doesn't have a route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import WrapperCenter from 'components/WrapperCenter';

/* eslint-disable react/prefer-stateless-function */
class NotFoundPage extends React.PureComponent {
  render() {
    return <WrapperCenter>Page Not found</WrapperCenter>;
  }
}

NotFoundPage.propTypes = {
  // history: PropTypes.object.isRequired,
  // intl: PropTypes.object.isRequired,
};

export default injectIntl(NotFoundPage);
