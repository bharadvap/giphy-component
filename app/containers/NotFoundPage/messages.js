/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  notFoundMessage: {
    id: 'app.containers.NotFound.notFoundMessage',
    defaultMessage: 'We canâ€™t seem to find the page you are looking for.',
  },
  notFoundHeadline: {
    id: 'app.containers.NotFound.notFoundHeadline',
    defaultMessage: 'Page not Found.',
  },
  back: {
    id: 'app.containers.NotFound.back',
    defaultMessage: 'Back to Previous Page',
  },
});
