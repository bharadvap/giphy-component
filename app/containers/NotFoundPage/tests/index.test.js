import React from 'react';
import { IntlProvider } from 'react-intl';
import { shallow } from 'enzyme';
import NotFoundPage from '../index';

describe('<NotFoundPage />', () => {
  it('should render the page message', () => {
    const renderedComponent = shallow(
      <IntlProvider locale="en">
        <NotFoundPage />
      </IntlProvider>,
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
