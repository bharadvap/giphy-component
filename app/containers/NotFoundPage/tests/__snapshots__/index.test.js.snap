// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should render the page message 1`] = `<InjectIntl(NotFound) />`;
